### PR TITLE
Update Keystone Scripts and Requirements

### PR DESCRIPTION
**Pull request by intlabs**

### CODE REVIEW

File: debian/Dockerfile [hunk 1]
```diff
@@ -12,11 +12,14 @@ ENV PATH=/virtualenv/bin:${PATH} \
 RUN set -x \
 	&& apt-get update \
 	&& apt-get install -y --no-install-recommends \
+# Project specific packages start
```
Comment (SamYaple) [inline debian/Dockerfile]: this was a change I was looking at making. +1

File: debian/Dockerfile [hunk 2]
```diff
@@ -34,13 +37,25 @@ RUN set -x \
 	&& hash -r \
 	&& pip install --no-compile --upgrade pip setuptools \
 	&& pip install --no-index --no-compile --find-links /tmp/packages --constraint /tmp/packages/upper-constraints.txt /tmp/${PROJECT} \
+	&& groupadd -g 42424 ${PROJECT} \
+	&& useradd -u 42424 -g ${PROJECT} -d /var/lib/${PROJECT} -s /sbin/nologin -c "${PROJECT} user" ${PROJECT} \
```
Comment (SamYaple) [inline debian/Dockerfile]: Should have -M to not create user-home

shell should be '/usr/sbin/nologin' I believe

File: debian/Dockerfile [hunk 3]
```diff
@@ -34,13 +37,25 @@ RUN set -x \
...
+	&& mkdir -p /etc/${PROJECT} /var/log/${PROJECT} /var/lib/${PROJECT} /var/cache/${PROJECT} \
+	&& chown ${PROJECT}:${PROJECT} /etc/${PROJECT} /var/log/${PROJECT} /var/lib/${PROJECT} /var/cache/${PROJECT} \
+# Project specific command block start
+	&& pip install --no-index --no-compile --find-links /tmp/packages --constraint /tmp/packages/upper-constraints.txt python-memcached pymysql \
+	&& rm /etc/apache2/ports.conf /etc/apache2/sites-enabled/* /etc/apache2/sites-available/* \
+	&& KS_WSGI_PUBLIC=$(which keystone-wsgi-public) \
+	&& KS_WSGI_ADMIN=$(which keystone-wsgi-admin) \
+	&& ln -s ${KS_WSGI_PUBLIC} /usr/bin/keystone-wsgi-public \
+	&& ln -s ${KS_WSGI_ADMIN} /usr/bin/keystone-wsgi-admin \
+	&& mkdir -p /var/www/cgi-bin/keystone \
```
Comment (SamYaple) [inline debian/Dockerfile]: are these lines really needed? do we have to link it to /var/www/cgi-bin?

File: debian/Dockerfile [hunk 4]
```diff
@@ -34,13 +37,25 @@ RUN set -x \
...
 	&& pip install --no-compile --upgrade pip setuptools \
 	&& pip install --no-index --no-compile --find-links /tmp/packages --constraint /tmp/packages/upper-constraints.txt /tmp/${PROJECT} \
+	&& groupadd -g 42424 ${PROJECT} \
+	&& useradd -u 42424 -g ${PROJECT} -d /var/lib/${PROJECT} -s /sbin/nologin -c "${PROJECT} user" ${PROJECT} \
+	&& mkdir -p /etc/${PROJECT} /var/log/${PROJECT} /var/lib/${PROJECT} /var/cache/${PROJECT} \
+	&& chown ${PROJECT}:${PROJECT} /etc/${PROJECT} /var/log/${PROJECT} /var/lib/${PROJECT} /var/cache/${PROJECT} \
+# Project specific command block start
+	&& pip install --no-index --no-compile --find-links /tmp/packages --constraint /tmp/packages/upper-constraints.txt python-memcached pymysql \
+	&& rm /etc/apache2/ports.conf /etc/apache2/sites-enabled/* /etc/apache2/sites-available/* \
+	&& KS_WSGI_PUBLIC=$(which keystone-wsgi-public) \
```
Comment (SamYaple) [inline debian/Dockerfile]: personally not liking the way this looks, setting the variable here. not a -1, just looks cludgy

File: debian/Dockerfile (hunk 2 above)
Comment (intlabs) [inline debian/Dockerfile]: will fix in next commit

**Pull request merged**